### DISCUSSION TITLE
Github Action on file update

### DISCUSSION THREAD

**Discussion by hanstra** [community/community#37983] [Actions]
I'm trying to use a github action to run a test prior to deployment. If I do this through a pull request, part of the pull request check runs the github action properly and I see that it either passed or did not pass the test.

However, if I update a file directly using the edit from GitHub and then commit the change, while the GitHub action does run, it does not test prior to commit. It just does the commit and there is no indication whether it passed the test or not.

Is there a way to add a check into a direct update after edits or can that only be done if there is a pull request involved?

**Comment by airtower-luna** [comment]
> Is there a way to add a check into a direct update after edits or can that only be done if there is a pull request involved?

Look at the [`push` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push), it will trigger whenever a branch changes (with the option to filter). It doesn't matter if it's by a commit via the web UI or `git push` on the command line. :slightly_smiling_face:

**Comment by david-engelmann** [comment]
If the action is design to run on the commit, you should see the workflow run from your actions section in the github UI
